Move monorepo scanning to KB and update verbiage (#1520)
semgrep/semgrep-docs: docs/kb/semgrep-ci/scan-monorepo-in-parts.md

> When scanning a repo with Semgrep in CI, the base command is `semgrep ci`. To understand this default setup for your source code manager (SCM) and CI provider, see [Getting started with Semgrep in continuous integration (CI)](/deployment/add-semgrep-to-ci).

Reviewer: Is this meant to be `repo`, or `monorepo`?

Author: It's meant to be repo since we are just talking about running a Semgrep scan in general here, not specific to monorepos.

You will then have the flexibility to trigger each one on appropriate events or frequencies.

> Now that you've successfully understood how to configure your monorepo to be scanned in parts, you also have to understand how to configure the findings from each part or module to show up as their own project in Semgrep AppSec Platform.

Reviewer: ```suggestion
Now that you understand how to configure your monorepo to be scanned in parts, you also have to understand how to configure the findings from each part or module to show up as their own project in Semgrep AppSec Platform.
```

> To assign findings from the module to their own project in Semgrep Appsec Platform, you must explicitly set the `SEMGREP_REPO_DISPLAY_NAME` environment variable ([see CI environment variables reference](/docs/semgrep-ci/ci-environment-variables#semgrep_repo_display_name)).

Reviewer 2: I see that in the link we mention the minimum version of Semgrep required - in case folks don't click through, I wonder if we should make a quick note of that here as well.

> To have findings from the module be assigned to their own project in Semgrep AppSec Platform, you will need to explicitly set the `SEMGREP_REPO_DISPLAY_NAME` environment variable ([see CI environment variables reference](/docs/semgrep-ci/ci-environment-variables#semgrep_repo_display_name)).

Reviewer: ```suggestion
To assign findings from the module to their own project in Semgrep Appsec Platform, you must explicitly set the `SEMGREP_REPO_DISPLAY_NAME` environment variable ([see CI environment variables reference](/docs/semgrep-ci/ci-environment-variables#semgrep_repo_display_name)).
```

Author: @s-santillan just curious why you suggested to change "AppSec" to "Appsec" here. I just realized everywhere else in the doc it's "AppSec" and Pablo's message about the name change also uses the capital S.

Author: I think this was a typo, changing back to capital S.

:::info

> Ensure that `SEMGREP_REPO_NAME` ([see reference](/docs/semgrep-ci/ci-environment-variables#semgrep_repo_name)) is still properly set as with any Semgrep scan, to ensure that hyperlink and PR/MR comment functionality is retained.

Author: ```suggestion
Ensure that `SEMGREP_REPO_NAME` is still properly set (either automatically if using a [supported SCM](/docs/category/pr-or-mr-comments) or [explicitly](/docs/semgrep-ci/ci-environment-variables#semgrep_repo_name)) as with any Semgrep scan, to ensure that hyperlink and PR/MR comment functionality is retained.
```

> You will need to ensure that `SEMGREP_REPO_NAME` ([see reference](/docs/semgrep-ci/ci-environment-variables#semgrep_repo_name)) is still being properly set as with any Semgrep scan, to ensure that hyperlink and PR/MR comment functionality is retained.

Reviewer: ```suggestion
Ensure that `SEMGREP_REPO_NAME` ([see reference](/docs/semgrep-ci/ci-environment-variables#semgrep_repo_name)) is still properly set as with any Semgrep scan, to ensure that hyperlink and PR/MR comment functionality is retained.
```

:::

> For example, if your monorepo is located at `https://github.com/semgrep/monorepo` the `SEMGREP_REPO_DISPLAY_NAME` would default to the value of `SEMGREP_REPO_NAME`, which in this case would be `semgrep/monorepo`. To split the monorepo into four projects corresponding to the logical modules, set `SEMGREP_REPO_NAME` as you normally would while setting `SEMGREP_REPO_DISPLAY_NAME` to something relevant before running Semgrep:

Author: I thought about that last sentence a while, I think just mentioning "relevant" is enough given all the previous context in the article.
